### DISCUSSION TITLE
Unify fixers related to blank lines before namespace

### DISCUSSION THREAD

**Discussion by Wirone** [PHP-CS-Fixer/PHP-CS-Fixer#7043] [RFC]
I think we could deprecate [`no_blank_lines_before_namespace`](https://cs.symfony.com/doc/rules/namespace_notation/no_blank_lines_before_namespace.html) and [`single_blank_line_before_namespace`](https://cs.symfony.com/doc/rules/namespace_notation/single_blank_line_before_namespace.html), and introduce `blank_line_before_namespace` with configurable strategies. This way discoverability would be better - having 2 separate fixers makes it harder to find that there are alternative options.

Related to #4897.

---

FYI: Done in #7053

**Comment by keradus** [comment]
@Wirone , this decision was already made, simply execution was never prioritised.

Ultimately, all fixers that are conflicting one to another should be changed [eg combined into single rule]. You can find them here:
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/b9a9bd67aa937e105bba853d998b61b55efdb42a/src/FixerFactory.php#L200-L210

**Reply by Wirone** [reply]
OK, so it seems like something to do 🙂. Thanks for pointing out this place in `FixerFactory`.

**Reply by keradus** [reply]
> simply execution was never prioritised.

OK, i will self-correct myself :D

we had gazillion of conflicting fixers in past, most of the job is done, simply some long-tail present
